primitive_benchmarks.rs fails to compile: StateCell::transition renamed to cas

## Description

`cargo bench --bench primitive_benchmarks -p strata-engine` fails to compile with two errors:

```
error[E0599]: no method named `transition` found for struct `StateCell`
   --> crates/engine/benches/primitive_benchmarks.rs:108:18

error[E0614]: type `i64` cannot be dereferenced
   --> crates/engine/benches/primitive_benchmarks.rs:110:42
```

## Root Cause

`StateCell::transition()` was renamed to `StateCell::cas()` but `primitive_benchmarks.rs` was not updated. The second error (`*n` on an `i64`) is also stale — `Value::Int` now contains `i64` directly, not `&i64`.

## Fix

In `crates/engine/benches/primitive_benchmarks.rs`, lines 107–116:

1. Replace `.transition(` with `.cas(` (and update the closure signature to match the new API)
2. Replace `*n` with `n` on line 110

## Affected Code

```rust
// Current (broken):
state_cell
    .transition(&branch_id, "bench_cell", |state| {
        let val = match &state.value {
            Value::Int(n) => *n,
            _ => 0,
        };
        Ok((Value::Int(val + 1), val + 1))
    })

// Needs updating to match current StateCell::cas API
```

--- a/crates/engine/benches/primitive_benchmarks.rs
+++ b/crates/engine/benches/primitive_benchmarks.rs
@@ -101,17 +101,24 @@ fn bench_state_cas(c: &mut Criterion) {
     let mut group = c.benchmark_group("state_cell");
     group.throughput(Throughput::Elements(1));
 
-    // Use transition which handles version automatically
+    // Read current version, then CAS with incremented value
     group.bench_function("cas", |b| {
         b.iter(|| {
+            let current = state_cell
+                .read_versioned(&branch_id, "bench_cell")
+                .unwrap()
+                .unwrap();
+            let val = match current.value {
+                Value::Int(n) => n,
+                _ => 0,
+            };
             state_cell
-                .transition(&branch_id, "bench_cell", |state| {
-                    let val = match &state.value {
-                        Value::Int(n) => *n,
-                        _ => 0,
-                    };
-                    Ok((Value::Int(val + 1), val + 1))
-                })
+                .cas(
+                    &branch_id,
+                    "bench_cell",
+                    current.version,
+                    Value::Int(val + 1),
+                )
                 .unwrap()
         })
     });

